### PR TITLE
stats: add built-in log linear histogram support 

### PR DESCRIPTION
**Pull request by ramaraochavali**
Description:
Implements built-in log linear histogram. Fixes the issue #1965

Risk Level: High

Testing:
Unit, Integration and Manual testing.

Docs Changes:
NA

Release Notes:
Will create PR for version update once this is approved.

### CODE REVIEW

File: source/server/http/admin.cc [hunk 1]
```diff
@@ -402,18 +402,34 @@ Http::Code AdminImpl::handlerStats(absl::string_view url, Http::HeaderMap& respo
     all_stats.emplace(gauge->name(), gauge->value());
   }
 
+  for (const Stats::ParentHistogramSharedPtr& histogram : server_.stats().histograms()) {
+    std::vector<std::string> summary;
+    const std::vector<double>& supported_quantiles_ref =
+        histogram->intervalStatistics().supportedQuantiles();
+    for (size_t i = 0; i < supported_quantiles_ref.size(); ++i) {
+      summary.push_back(fmt::format("P{}({},{})", 100 * supported_quantiles_ref[i],
```
Comment (ramaraochavali) [inline source/server/http/admin.cc]: @mattklein123 This is one of the points you had concerns last review about displaying the histogram information in single line.
Here is how it looks with the latest change
`cluster.time.upstream_rq_time: P0(0,0) P25(0,0) P50(0,0) P75(0,1.015) P90(1.06387,1.07194) P95(1.08525,1.09093) P99(2.04331,2.07759) P99.9(4.08333,23) P100(23,140)`
First value is interval value and second value is cumulative value for each quantile.

**Comment by ramaraochavali** [comment]
@mattklein123 @jmarantz suggested to squash all commits in to one as more number of commits may be causing the PR not to load. I tried rebase and squash but because of the repo reorg PRs that I merged in to my branch i am getting in to some trouble. He suggested another approach which I will look in to tomorrow to see if I can get a clean PR with all the commits squashed in to one. 

Would you mind taking a look at the test cases at this stage?

**Comment by jmarantz** [comment]
per discussion in https://github.com/envoyproxy/envoy/issues/3125 the number of commits is related to (though not the sole cause of) the PRs blowing up, so this PR may blow up too.  Hence my suggestion to make a clean PR with these changes.

**Comment by ramaraochavali** [comment]
@jmarantz sure. some of my commits are on old repo structure and some of them are on new. So both rebase and patch are failing to merge them together..will try again tomorrow.

**Comment by jmarantz** [comment]
You definitely have to merge master before you can follow the steps I suggested.  Then it should be 2 easy steps.

File: source/server/http/admin.cc (hunk 1 above)
Comment (mattklein123) [reply source/server/http/admin.cc]: Do we think cumulative has much benefit? I can't imagine it being very useful from a debugging perspective so I wonder if we should just drop it here? Thoughts?

**Comment by mattklein123** [comment]
@ramaraochavali I agree with @jmarantz . I think we are very close, but I'm going to give this a super thorough review anyway, so let's just start completely from scratch. I would just do a completely fresh PR with a single commit. And then we can do final review?

**Pull request closed without merging**